### PR TITLE
[AAP-20127] Add event streams

### PR DESCRIPTION
**Pull request by lgalis**
Remove Event sources, add Event stream pages

Screenshot from 2024-02-12 14-50-32
Screenshot from 2024-02-12 14-27-47
Screenshot from 2024-02-08 14-21-53
Screenshot from 2024-02-08 14-21-56
Screenshot from 2024-02-08 14-22-01
Screenshot from 2024-02-08 14-22-18
Screenshot from 2024-02-08 14-22-25

### CODE REVIEW

File: frontend/eda/access/credentials/components/PageFormCredentialsSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      labelHelpTitle={t('Credentials')}
+      labelHelp={props.labelHelp}
+      label={t('Credential')}
+      selectTitle={t('Select a credential')}
+      selectOpen={selectCredential}
+      validate={(credentials: EdaCredential[]) => {
+        if (props.isRequired && credentials.length === 0) {
+          return t('Credential is required.');
+        }
+        return undefined;
```
Comment (akus062381) [inline frontend/eda/access/credentials/components/PageFormCredentialsSelect.tsx:29]: can we get a data-cy added to this section?

File: frontend/eda/access/credentials/EditCredential.tsx [hunk 2]
```diff
@@ -75,21 +82,48 @@ function CredentialInputs() {
...
+        />
+        <PageFormTextInput<EdaCredentialCreate>
+          name="secret"
+          label={t('Vault password')}
+          type="password"
+          placeholder={t('Enter vault password')}
+          isRequired
+          labelHelp={t('Vault password')}
+          labelHelpTitle={t('Vault password')}
+        />
```
Comment (akus062381) [inline frontend/eda/access/credentials/EditCredential.tsx:125]: Can we get data-cy locators added to the elements in this file that we may need to be able to access with Cypress?

File: frontend/eda/event-streams/EventStreamForm.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        submitText={t('Create event stream')}
+        onSubmit={onSubmit}
+        cancelText={t('Cancel')}
+        onCancel={onCancel}
+        defaultValue={{
+          restart_policy: RestartPolicyEnum.OnFailure,
+          is_enabled: true,
+          source_args: '',
+        }}
+      >
```
Comment (akus062381) [inline frontend/eda/event-streams/EventStreamForm.tsx:61]: data-cy here?

File: frontend/eda/event-streams/EventStreamForm.tsx [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+          labelHelp={t(
+            `The arguments for the rulebook are in a JSON or YAML format. 
+            The content would be equivalent to the file passed through the '--vars' flag of ansible-rulebook command.`
+          )}
+          labelHelpTitle={t('Arguments')}
+        />
+      </PageFormSection>
+    </>
+  );
+}
```
Comment (akus062381) [inline frontend/eda/event-streams/EventStreamForm.tsx:163]: can we have data-cys added to the above elements here?

File: frontend/eda/event-streams/EventStreamInstancePage/EventStreamInstanceDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        </PageDetail>
+      </PageDetails>
+      <PageDetailsSection>
+        {eventStreamInstanceLog?.results?.length ? (
+          <PageDetailCodeEditor
+            label={t('Output')}
+            value={eventStreamInstanceLog?.results?.map((item) => item.log).join('\r\n')}
+            showCopyToClipboard={true}
+          />
+        ) : null}
```
Comment (akus062381) [inline frontend/eda/event-streams/EventStreamInstancePage/EventStreamInstanceDetails.tsx:55]: can we have data-cys added to the above elements here?

File: frontend/eda/event-streams/EventStreamInstancePage/EventStreamInstancePage.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+              params: { id: eventStream?.id },
+            }),
+          },
+          { label: `${eventStreamInstance?.id || ''} - ${eventStreamInstance?.name || ''}` },
+        ]}
+      />
+      <PageRoutedTabs
+        tabs={[{ label: t('Details'), page: EdaRoute.EventStreamInstanceDetails }]}
+        params={{ id: eventStreamInstance?.event_stream_id, instanceId: params.instanceId }}
+      />
```
Comment (akus062381) [inline frontend/eda/event-streams/EventStreamInstancePage/EventStreamInstancePage.tsx:45]: can we have data-cys added to the above elements here?

File: frontend/eda/event-streams/EventStreamPage/EventStreamDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+          <PageDetailCodeEditor
+            value={eventStream.source_args}
+            showCopyToClipboard={true}
+            label={t('Arguments')}
+            helpText={t('Arguments')}
+          />
+        </PageSection>
+      )}
+    </Scrollable>
+  );
```
Comment (akus062381) [inline frontend/eda/event-streams/EventStreamPage/EventStreamDetails.tsx:96]: can we have data-cys added to the above elements here?

File: frontend/eda/event-streams/EventStreamPage/EventStreamHistory.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      <PageTable
+        tableColumns={tableColumns}
+        toolbarFilters={toolbarFilters}
+        errorStateTitle={t('Error loading history')}
+        emptyStateTitle={t('No event stream history')}
+        emptyStateIcon={CubesIcon}
+        emptyStateDescription={t('No history for this event stream')}
+        {...view}
+        defaultSubtitle={t('Event Stream History')}
+      />
```
Comment (akus062381) [inline frontend/eda/event-streams/EventStreamPage/EventStreamHistory.tsx:34]: can we have data-cys added to the above elements here?

File: frontend/eda/event-streams/EventStreamPage/EventStreamPage.tsx [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+            type: PageActionType.Seperator,
+          },
+          {
+            type: PageActionType.Button,
+            selection: PageActionSelection.Single,
+            icon: TrashIcon,
+            label: t('Delete event stream'),
+            onClick: (eventStream: EdaEventStream) => deleteEventStreams([eventStream]),
+            isDanger: true,
+          },
```
Comment (akus062381) [inline frontend/eda/event-streams/EventStreamPage/EventStreamPage.tsx:121]: can we have data-cys added to the above elements here?

File: frontend/eda/event-streams/EventStreams.tsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+        toolbarActions={toolbarActions}
+        toolbarFilters={toolbarFilters}
+        rowActions={rowActions}
+        errorStateTitle={t('Error loading event streams')}
+        emptyStateTitle={t('There are currently no event streams created for your organization.')}
+        emptyStateDescription={t('Please create a event stream by using the button below.')}
+        emptyStateButtonText={t('Create event stream')}
+        emptyStateButtonClick={() => pageNavigate(EdaRoute.CreateEventStream)}
+        {...view}
+        defaultSubtitle={t('Event stream')}
```
Comment (akus062381) [inline frontend/eda/event-streams/EventStreams.tsx:39]: data-cys here?

File: frontend/eda/event-streams/hooks/useControlEventStreams.tsx [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+        keyFn: (item) => item?.id,
+        isDanger: true,
+        confirmationColumns,
+        actionColumns,
+        onComplete,
+        actionFn: (eventStream: EdaEventStream) =>
+          postRequest(edaAPI`/event-streams/${eventStream.id.toString()}/disable/`, undefined),
+      });
+    },
+    [actionColumns, bulkAction, confirmationColumns, postRequest, onComplete, t]
```
Comment (akus062381) [inline frontend/eda/event-streams/hooks/useControlEventStreams.tsx:68]: data-cys here?

File: frontend/eda/event-streams/hooks/useControlEventStreams.tsx [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+        actionButtonText: t('Restart event streams', { count: eventStreams.length }),
+        items: eventStreams.sort((l, r) => compareStrings(l.name, r.name)),
+        keyFn: (item) => item?.id,
+        confirmationColumns,
+        actionColumns,
+        onComplete,
+        actionFn: (eventStream: EdaEventStream) =>
+          postRequest(edaAPI`/event-streams/${eventStream.id.toString()}/restart/`, undefined),
+      });
+    },
```
Comment (akus062381) [inline frontend/eda/event-streams/hooks/useControlEventStreams.tsx:94]: data-cys here?

File: frontend/eda/event-streams/hooks/useDeleteEventStreams.tsx [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+        actionButtonText: t('Delete event streams', { count: eventStreams.length }),
+        items: eventStreams.sort((l, r) => compareStrings(l.name, r.name)),
+        keyFn: (item) => item?.id,
+        isDanger: true,
+        confirmationColumns,
+        actionColumns,
+        onComplete,
+        actionFn: (eventStream: EdaEventStream, signal) =>
+          requestDelete(edaAPI`/event-streams/${eventStream.id.toString()}/`, signal),
+      });
```
Comment (akus062381) [inline frontend/eda/event-streams/hooks/useDeleteEventStreams.tsx:31]: data-cys here?

File: frontend/eda/event-streams/hooks/useEventStreamActions.tsx [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+        type: PageActionType.Button,
+        selection: PageActionSelection.Single,
+        icon: TrashIcon,
+        label: t('Delete event stream'),
+        isHidden: (eventStream: EdaEventStream) => eventStream?.status === Status906Enum.Deleting,
+        onClick: (eventStream: EdaEventStream) => deleteEventStreams([eventStream]),
+        isDanger: true,
+      },
+    ];
+    return actions;
```
Comment (akus062381) [inline frontend/eda/event-streams/hooks/useEventStreamActions.tsx:89]: can we have data-cys added to the above elements here?

File: frontend/eda/event-streams/hooks/useEventStreamColumns.tsx [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+        value: (eventStream) => eventStream.modified_at,
+        table: ColumnTableOption.expanded,
+        card: 'hidden',
+        list: 'secondary',
+        modal: 'hidden',
+        dashboard: 'hidden',
+      },
+    ],
+    [getPageUrl, t]
+  );
```
Comment (akus062381) [inline frontend/eda/event-streams/hooks/useEventStreamColumns.tsx:89]: can we have data-cys added to the above elements here?

File: frontend/eda/event-streams/hooks/useEventStreamHistoryActions.tsx [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+      {
+        type: PageActionType.Button,
+        selection: PageActionSelection.Multiple,
+        icon: TrashIcon,
+        label: t('Delete selected event streams'),
+        onClick: (eventStreams: EdaEventStream[]) => deleteEventStreams(eventStreams),
+      },
+    ],
+    [deleteEventStreams, pageNavigate, t]
+  );
```
Comment (akus062381) [inline frontend/eda/event-streams/hooks/useEventStreamHistoryActions.tsx:39]: can we have data-cys added to the above elements here?

File: frontend/eda/event-streams/hooks/useEventStreamHistoryFilters.tsx [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+          { label: t('Pending'), value: 'pending' },
+          { label: t('Failed'), value: 'failed' },
+          { label: t('Stopping'), value: 'stopping' },
+          { label: t('Stopped'), value: 'stopped' },
+          { label: t('Completed'), value: 'completed' },
+          { label: t('Unresponsive'), value: 'unresponsive' },
+        ],
+        placeholder: t('Select statuses'),
+      },
+    ],
```
Comment (akus062381) [inline frontend/eda/event-streams/hooks/useEventStreamHistoryFilters.tsx:26]: can we have data-cys added to the above elements here?

File: frontend/eda/event-streams/hooks/useEventStreamsActions.tsx [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+        isDanger: true,
+      },
+    ];
+    return actions;
+  }, [
+    deleteEventStreams,
+    enableEventStreams,
+    disableEventStreams,
+    restartEventStreams,
+    pageNavigate,
```
Comment (akus062381) [inline frontend/eda/event-streams/hooks/useEventStreamsActions.tsx:112]: can we have data-cys added to the above elements here?

**Review by akus062381 — CHANGES_REQUESTED** [review]

File: frontend/eda/event-streams/hooks/useEventStreamActions.tsx (hunk 14 above)
Comment (lgalis) [reply frontend/eda/event-streams/hooks/useEventStreamActions.tsx:89]: the component used here already adds a data-cy for each menu item

File: frontend/eda/event-streams/hooks/useEventStreamColumns.tsx (hunk 15 above)
Comment (lgalis) [reply frontend/eda/event-streams/hooks/useEventStreamColumns.tsx:89]: same as above - data-cy are added by the component

File: frontend/eda/event-streams/hooks/useEventStreamHistoryActions.tsx (hunk 16 above)
Comment (lgalis) [reply frontend/eda/event-streams/hooks/useEventStreamHistoryActions.tsx:39]: same as above -the component adds data-cy s

File: frontend/eda/event-streams/hooks/useEventStreamActions.tsx (hunk 14 above)
Comment (lgalis) [reply frontend/eda/event-streams/hooks/useEventStreamActions.tsx:89]: data-cy added by component

File: frontend/eda/event-streams/hooks/useEventStreamColumns.tsx (hunk 15 above)
Comment (lgalis) [reply frontend/eda/event-streams/hooks/useEventStreamColumns.tsx:89]: data-cy added by component

File: frontend/eda/event-streams/hooks/useEventStreamHistoryActions.tsx (hunk 16 above)
Comment (lgalis) [reply frontend/eda/event-streams/hooks/useEventStreamHistoryActions.tsx:39]: data-cy added by component

File: frontend/eda/event-streams/hooks/useEventStreamHistoryFilters.tsx (hunk 17 above)
Comment (lgalis) [reply frontend/eda/event-streams/hooks/useEventStreamHistoryFilters.tsx:26]: data-cy added by component

File: frontend/eda/event-streams/hooks/useEventStreamsActions.tsx (hunk 18 above)
Comment (lgalis) [reply frontend/eda/event-streams/hooks/useEventStreamsActions.tsx:112]: data-cy added by component

File: frontend/eda/event-streams/hooks/useControlEventStreams.tsx (hunk 11 above)
Comment (lgalis) [reply frontend/eda/event-streams/hooks/useControlEventStreams.tsx:68]: data-cy added by component

File: frontend/eda/event-streams/hooks/useDeleteEventStreams.tsx (hunk 13 above)
Comment (lgalis) [reply frontend/eda/event-streams/hooks/useDeleteEventStreams.tsx:31]: data-cy added by component

File: frontend/eda/event-streams/hooks/useControlEventStreams.tsx (hunk 12 above)
Comment (lgalis) [reply frontend/eda/event-streams/hooks/useControlEventStreams.tsx:94]: data-cy added by component

File: frontend/eda/event-streams/EventStreamPage/EventStreamDetails.tsx (hunk 7 above)
Comment (lgalis) [reply frontend/eda/event-streams/EventStreamPage/EventStreamDetails.tsx:96]: data-cy added by the components used

File: frontend/eda/event-streams/EventStreams.tsx (hunk 10 above)
Comment (lgalis) [reply frontend/eda/event-streams/EventStreams.tsx:39]: data-cy already exists - provided by components

File: frontend/eda/event-streams/EventStreamPage/EventStreamPage.tsx (hunk 9 above)
Comment (lgalis) [reply frontend/eda/event-streams/EventStreamPage/EventStreamPage.tsx:121]: data-cy already exists - provided by components

File: frontend/eda/event-streams/EventStreamPage/EventStreamHistory.tsx (hunk 8 above)
Comment (lgalis) [reply frontend/eda/event-streams/EventStreamPage/EventStreamHistory.tsx:34]: data-cy provide by components

**Comment by lgalis** [comment]
@akus062381 - comments addressed

**Review by akus062381 — APPROVED** [review]
awesome work, thank you!

**Pull request merged**